### PR TITLE
MongoDB Connector

### PR DESCRIPTION
**Pull request by DianaMDB**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @DianaMDB! Please check comments after the review, thank you!
Let me know if you have any questions!

**Comment by v-EgorKozhadei** [comment]
Hello @DianaMDB,

I hope you are doing well. 

I am following up with you to see how things are going with preparing your connector. 

Please let us know if you have any questions. 

Thank you.

**Comment by DianaMDB** [comment]
Hi @v-EgorKozhadei ,
      Thanks for the follow up.
We have made the changes as suggested and are in the last stages of re-testing as response codes are also being added.

We will resubmit the definition and properties files tomorrow.

Thanks for your support.
Best Regards,




{ Name     : “Diana Annie Jenosh",
  Title    : "Senior Solutions Architect",
  Phone    : "+ 91 8861929054",
  Location : "Bengaluru, India"}

**Comment by DianaMDB** [comment]
Hi @v-EgorKozhadei , Team,
       Have made the required changes and uploaded the updated properties file and swagger definition.
Request you to kindly review and let us know if any concerns.

Thanks as always!
Best Regards,




{ Name     : “Diana Annie Jenosh",
  Title    : "Senior Solutions Architect",
  Phone    : "+ 91 8861929054",
  Location : "Bengaluru, India"}

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,1421 @@
...
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "dataSource": {
+                  "type": "string",
+                  "description": "datasource",
+                  "title": ""
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:67]: @DianaMDB please add value for this property. "Data source" would be OK.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:131]: @DianaMDB empty schemas are [not allowed](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#define-exact-operation-responses).

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,1421 @@
...
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/action/insertOne": {
+      "post": {
+        "summary": "Insert Document",
+        "description": "Using the insertone endpoint, you can add a document to a collection.Use the document property in the request body to specify the document that should be created.",
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:28]: @DianaMDB please make sure that everything is grammatically correct and is free of typographical errors.
Do you want to keep `insertone` as a single word? Should it have casing if it's the name of the command?
Also please make sure that there are whitespaces after commas and points.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "500": {
+            "description": "Internal Server Error",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:145]: @DianaMDB empty schemas are [not allowed](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#define-exact-operation-responses).

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:284]: @DianaMDB empty schemas are [not allowed](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#define-exact-operation-responses).

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "500": {
+            "description": "Internal Server Error",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:298]: @DianaMDB empty schemas are [not allowed](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#define-exact-operation-responses).

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:441]: @DianaMDB mentioned in the comments above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "500": {
+            "description": "Internal Server Error",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:455]: @DianaMDB mentioned in the comments above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 9]
```diff
@@ -0,0 +1,1421 @@
...
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "dataSource": {
+                  "type": "string",
+                  "description": "datasource",
+                  "title": ""
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:512]: @DianaMDB please look through empty titles and add value to them. Value should be a valid sentence, e.g. `Data source`.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 10]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:571]: @DianaMDB mentioned above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 11]
```diff
@@ -0,0 +1,1421 @@
...
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "dataSource": {
+                  "type": "string",
+                  "description": "datasource",
+                  "title": ""
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:642]: @DianaMDB mentioned above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 12]
```diff
@@ -0,0 +1,1421 @@
...
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "dataSource": {
+                  "type": "string",
+                  "description": "datasource",
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:641]: @DianaMDB make sure descriptions are grammatically correct and free of typographical errors. You can refer to [English strings guidelines](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#create-quality-english-language-strings).

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 13]
```diff
@@ -0,0 +1,1421 @@
...
+                  "title": ""
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:802]: @DianaMDB mentioned above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 14]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:887]: @DianaMDB mentioned above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 15]
```diff
@@ -0,0 +1,1421 @@
...
+                  }
+                }
+              }
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A JSON object containing array of all the documents fetched",
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:846]: @DianaMDB make sure everything is grammatically correct, you can refer to [English strings guidelines](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#create-quality-english-language-strings).

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 16]
```diff
@@ -0,0 +1,1421 @@
...
+          {
+            "name": "body",
+            "in": "body",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "dataSource": {
+                  "type": "string",
+                  "description": "datasource",
+                  "title": ""
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:958]: @DianaMDB mentioned above.

File: certified-connectors/MongoDB/apiDefinition.swagger.json [hunk 17]
```diff
@@ -0,0 +1,1421 @@
...
+              },
+              "Accept": {
+                "description": "Accept",
+                "type": "string"
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {},
```
Comment (v-EgorKozhadei) [inline certified-connectors/MongoDB/apiDefinition.swagger.json:1361]: @DianaMDB empty schema, mentioned above.

**Review by v-EgorKozhadei — COMMENTED** [review]
Hi @DianaMDB! Please check comments after the review. Let me know if you have any questions.
Thank you!

File: certified-connectors/MongoDB/apiDefinition.swagger.json (hunk 3 above)
Comment (v-EgorKozhadei) [reply certified-connectors/MongoDB/apiDefinition.swagger.json:28]: Hi @DianaMDB!
You can keep your commands in camel case, e.g. `InsertOne` or `insertOne`.
Camel case is not suitable for string localization, but it's fine for command names if you want to keep them this way.

File: certified-connectors/MongoDB/apiDefinition.swagger.json (hunk 15 above)
Comment (DianaMDB) [reply certified-connectors/MongoDB/apiDefinition.swagger.json]: Have changed "containing array" to "containing an array".

**Comment by DianaMDB** [comment]
Hi Team,
     Have made the changes suggested. Kindly review and let us know.
Thanks as always.

Best Regards,




{ Name     : “Diana Annie Jenosh",
  Title    : "Senior Solutions Architect",
  Phone    : "+ 91 8861929054",
  Location : "Bengaluru, India"}

**Review by v-EgorKozhadei — APPROVED** [review]

**Comment by v-EgorKozhadei** [comment]
Hello @DianaMDB, 

I hope you are doing well. 

Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) using your latest commit id, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file. 

After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6. 

Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.  

We expect all tests to be completed within 1-2 weeks. 

If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions. 

Please let me know if you have any questions. 

Thank you very much for working with us.

**Comment by DianaMDB** [comment]
Hi @v-EgorKozhadei , Amjed, Connector Certification Team,

	   Many Thanks to you and your team for all the reviews and support in getting the corrections applied.
We will follow the next steps and let you know if any concerns.

 
Thanks again.

Best Regards,



{ Name     : “Diana Annie Jenosh",
  Title    : "Senior Solutions Architect",
  Phone    : "+ 91 8861929054",
  Location : "Bengaluru, India"}

File: certified-connectors/MongoDB/Readme.md [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+Read the full documentation of the operations and their examples [here](https://www.mongodb.com/docs/atlas/api/data-api-resources/#data-api-resources)
+
+## Getting Started
+1. In the Base URL field in the General tab, replace the placeholder **"_Data API App ID_"** with the App ID from the URL endpoint under **Data API** tab in your Atlas cluster.  
+2. For some APIs which expect a Json request or response , the layout/sample of the json needs to be added using the **"Import from Sample"** option for the Request or Response under the **Definition** tab.
+
+## Known Limitations
+
+1. Data API logs all requests and stores the logs for 30 days.They can be viewed from the **Data API** screen in the **Logs** tab
```
Comment (drewdipalma) [inline certified-connectors/MongoDB/Readme.md:37]: Note - Log retention is 10 days currently.

**Pull request merged**